src/site/content/en/blog/speedy-css-tip-animated-gradient-text/index.md

```diff
@@ -0,0 +1,168 @@
+date: 2022-11-03
+hero: image/Dyx9FwYgMyNqy1kMGx8Orz6q0qC3/jGDpMMoAas6YfIUAbPvr.jpg
+thumbnail: image/Dyx9FwYgMyNqy1kMGx8Orz6q0qC3/jGDpMMoAas6YfIUAbPvr.jpg
+alt: A hand holds up a sticker with the word "Hello!" printed on it. The text is orange and the background is black.
+authors:
+  - jheyy
+tags: 
+  - blog
+  - css
+---
+
+Hop over to [CodePen](https://pen.new) and create a new pen.
+
+Create the markup for your text. Let's use a header with the word "Speedy":
+
+```html
+<h1 class="boujee-text">Hello!</h1>
+```
+
+Then, let's give our `body` a darker `background-color`:
+
+```css
+body {
+  display: grid;
+  place-items: center;
+  min-height: 100vh;
+  background: hsl(0 0% 20%);
+}
+```
+
+Bump up the `font-size` on our text. Use `clamp` to make it responsive:
+
+```css
+.boujee-text {
+  font-size: clamp(3rem, 25vmin, 12rem);
+}
+```
+
+Create two custom properties for the colors we're going to use. Apply a `linear-gradient` to the `background` using those colors and rotate it by 90 degrees:
+
+```css/1-2,6-8
+.boujee-text {
+  --color-one: hsl(15 90% 55%);
+  --color-two: hsl(40 95% 55%);
+  font-size: clamp(3rem, 25vmin, 8rem);
+  background: linear-gradient(
+                90deg,
+                var(--color-one),
+                var(--color-two),
+                var(--color-one)
+              ) 0 0 / 100% 100%;
+}
+```
+
+Create a custom property that you can use for the horizontal background size. Use it for `background-size-x`:
+
+```css/1,10
+.boujee-text {
+  --bg-size: 400%;
+  --color-one: hsl(15 90% 55%);
+  --color-two: hsl(40 95% 55%);
+  font-size: clamp(3rem, 25vmin, 8rem);
+  background: linear-gradient(
+                90deg,
+                var(--color-one),
+                var(--color-two),
+                var(--color-one)
+              ) 0 0 / var(--bg-size) 100%;
+}
+```
+
+To clip the background to the text set the `color` to `transparent`, and set `background-clip: text`:
+
+
+```css/11-13
+.boujee-text {
+  --bg-size: 400%;
+  --color-one: hsl(15 90% 55%);
+  --color-two: hsl(40 95% 55%);
+  font-size: clamp(3rem, 25vmin, 8rem);
+  background: linear-gradient(
+                90deg,
+                var(--color-one),
+                var(--color-two),
+                var(--color-one)
+              ) 0 0 / var(--bg-size) 100%;
+  color: transparent;
+  -webkit-background-clip: text;
+  background-clip: text;
+}
+```
+
+{% Aside %}
+The `background-clip` property needs prefixing in some browsers as `-webkit-background-clip`.
+{% endAside %}
+
+At this point, you could leave it there and experiment with the `background-position` and the gradient used in the `background-image`. Or, you could animate that gradient across your text. First, define a keyframe for the animation. This will use the `--bg-size` custom property we defined earlier. A good example of scoped custom properties making maintenance easier for you.
```

Comment:
```suggestion
At this point, you could leave it there and experiment with the `background-position` and the gradient used in the `background-image`. Or, you could animate that gradient across your text. First, define a keyframe for the animation. This will use the `--bg-size` custom property you defined earlier. A good example of scoped custom properties making maintenance easier for you.
```